components/script/bluetooth_blacklist.rs

```diff
@@ -5,115 +5,123 @@
+            })
+        },
+        Blacklist::Reads => {
+            BLUETOOTH_BLACKLIST.with(|blist| {
+                blist.borrow().is_blacklisted_for_reads(uuid)
+            })
+        },
+        Blacklist::Writes => {
+            BLUETOOTH_BLACKLIST.with(|blist| {
+                blist.borrow().is_blacklisted_for_writes(uuid)
+            })
+        },
     }
 }
 
-pub struct BluetoothBlacklist(Option<HashMap<String, ExcludeType>>);
+pub struct BluetoothBlacklist(Option<HashMap<String, Blacklist>>);
 
 #[derive(Eq, PartialEq)]
-pub enum ExcludeType {
-    Exclude, // Read and Write
-    ExcludeReads,
-    ExcludeWrites,
+pub enum Blacklist {
+    All, // Read and Write
+    Reads,
+    Writes,
 }
 
 impl BluetoothBlacklist {
-
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted
     pub fn is_blacklisted(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All)),
             None => false,
         }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-reads
     pub fn is_blacklisted_for_reads(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
-                                                              et.eq(&ExcludeType::ExcludeReads)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All) ||
+                                                              et.eq(&Blacklist::Reads)),
             None => false,
         }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-writes
     pub fn is_blacklisted_for_writes(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
-                                                              et.eq(&ExcludeType::ExcludeWrites)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All) ||
+                                                              et.eq(&Blacklist::Writes)),
             None => false,
         }
     }
 }
 
 // https://webbluetoothcg.github.io/web-bluetooth/#parsing-the-blacklist
-fn parse_blacklist() -> Option<HashMap<String, ExcludeType>> {
+fn parse_blacklist() -> Option<HashMap<String, Blacklist>> {
     // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.
-    let file = match File::open(FILE_PATH) {
-        Ok(file) => file,
-        Err(_) => return None,
-    };
-    let content = BufReader::new(file);
     let valid_uuid_regex = Regex::new(VALID_UUID_REGEX).unwrap();
+    let mut data: Vec<u8> = Vec::new();
+    let lines = read_resource_file_lines("gatt_blacklist.txt", &mut data)
+                    .expect("Could not find gatt_blacklist.txt file");
```

Comment:
And i still think that we should use `const string` instead of the inline stings.